apps/omg_child_chain/lib/omg_child_chain/gas_price/strategy/block_percentile_gas_strategy.ex

```diff
@@ -0,0 +1,157 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.ChildChain.GasPrice.Strategy.BlockPercentileGasStrategy do
+  @moduledoc """
+  Suggests gas prices based on the percentile of blocks that accepted as the minimum gas price.
+
+  Requires `OMG.ChildChain.GasPrice.Strategy.History` running.
+  """
+  use GenServer
+  require Logger
+  alias OMG.ChildChain.GasPrice.History
+  alias OMG.ChildChain.GasPrice.Strategy
+
+  @type t() :: %__MODULE__{
+          prices:
+            %{
+              safe_low: float(),
+              standard: float(),
+              fast: float(),
+              fastest: float()
+            }
+            | error()
+        }
+
+  @type error() :: {:error, :no_gas_price_history}
+
+  defstruct prices: %{
+              safe_low: 20_000_000_000,
+              standard: 20_000_000_000,
+              fast: 20_000_000_000,
+              fastest: 20_000_000_000
+            }
+
+  @thresholds %{
+    safe_low: 35,
+    standard: 60,
+    fast: 90,
+    fastest: 100
+  }
+
+  @target_threshold :fast
+
+  @behaviour Strategy
+
+  @doc """
+  Starts the block percentile strategy.
+  """
+  @spec start_link(Keyword.t()) :: GenServer.on_start()
+  def start_link(init_arg) do
+    GenServer.start_link(__MODULE__, init_arg, name: __MODULE__)
+  end
+
+  @doc """
+  Suggests the optimal gas price.
+  """
+  @impl Strategy
+  @spec get_price() :: {:ok, pos_integer()} | error()
+  def get_price() do
+    GenServer.call(__MODULE__, :get_price)
+  end
+
+  @doc """
+  A stub that handles the recalculation trigger.
+
+  Since this strategy recalculates based on its interval and not a recalculation
+  triggered by the `recalculate/1`'s caller, this function simply returns `:ok` without any computation.
+
+  To get the price, use `get_price/0` instead.
+  """
+  @impl Strategy
+  @spec recalculate(Keyword.t()) :: :ok
+  def recalculate(_params) do
+    :ok
+  end
+
+  #
+  # GenServer initialization
+  #
+
+  @doc false
+  @impl GenServer
+  def init(_init_arg) do
+    _ = History.subscribe(self())
+    state = %__MODULE__{}
+
+    _ = Logger.info("Started #{__MODULE__}: #{inspect(state)}")
+    {:ok, state}
+  end
+
+  #
+  # GenServer callbacks
+  #
+
+  @doc false
+  @impl GenServer
+  def handle_call(:get_price, _, state) do
+    price =
+      case state.prices do
+        {:error, _} = error ->
+          error
+
+        prices ->
+          {:ok, prices[@target_threshold]}
+      end
+
+    {:reply, price, state}
+  end
+
+  @doc false
+  @impl GenServer
+  def handle_info({History, :updated}, state) do
+    prices = do_recalculate()
+
+    _ = Logger.info("#{__MODULE__}: History updated. Prices recalculated to: #{inspect(prices)}")
+    {:noreply, %{state | prices: prices}}
+  end
+
+  #
+  # Internal implementations
+  #
+
+  defp do_recalculate() do
```

Comment:
perhaps a quick desc in what it does.
I believe History.all() returns all stored blocks (like 200) and you find the median price?
What happens if you don't have enough blocks to properly estimate, could it even happen?